### ISSUE TITLE
[Feature]: Please added DPM++ 2M Karras v2

### ISSUE THREAD

**Issue by Satriouz** [vladmandic/automatic#2028]
### Feature description

----------------------------------------------------
edit file ..\repositories\k-diffusion\k_diffusion\sampling.py

at the end of file add:

@torch.no_grad()
def sample_dpmpp_2mV2(model, x, sigmas, extra_args=None, callback=None, disable=None):
    """DPM-Solver++(2M) V2."""
    extra_args = {} if extra_args is None else extra_args
    s_in = x.new_ones([x.shape[0]])
    sigma_fn = lambda t: t.neg().exp()
    t_fn = lambda sigma: sigma.log().neg()
    old_denoised = None

    for i in trange(len(sigmas) - 1, disable=disable):
        denoised = model(x, sigmas[i] * s_in, **extra_args)
        if callback is not None:
            callback({'x': x, 'i': i, 'sigma': sigmas[i], 'sigma_hat': sigmas[i], 'denoised': denoised})
        t, t_next = t_fn(sigmas[i]), t_fn(sigmas[i + 1])
        h = t_next - t
        if old_denoised is None or sigmas[i + 1] == 0:
            x = (sigma_fn(t_next) / sigma_fn(t)) * x - (-h).expm1() * denoised
        else:
            h_last = t - t_fn(sigmas[i - 1])
            r = h_last / h
            denoised_d = (1 + 1 / (2 * r)) * denoised - (1 / (2 * r)) * old_denoised
            x = (sigma_fn(t_next) / sigma_fn(t)) * x - (-h).expm1() * denoised_d
        #old_denoised = denoised
        #test new sampler
        sigma_progress = i / len(sigmas) 
        adjustment_factor = 1 + (0.15 * (sigma_progress * sigma_progress)) 
        old_denoised = denoised * adjustment_factor
    return x
Edit ..\modules\sd_samplers_kdiffusion.py

add

('DPM++ 2M Karras v2', 'sample_dpmpp_2mV2', ['k_dpmpp_2m_ka'], {'scheduler': 'karras'}),

after (around line 20- keep it)
----------------------------------------------------

**Comment by vladmandic** [comment]
directly editing k-diffusion is not advised as its pulled from its original upstream, so this should be added to source.
also, what is the origin of this sampler? paper? results?

**Comment by FirstZero** [comment]
@vladmandic To my understanding this change is related to https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/8457

**Comment by Satriouz** [comment]
> Mengedit langsung K-difusi tidak disarankan karena ditarik dari hulu aslinya, jadi ini harus ditambahkan ke sumber. Juga, apa asal usul sampler ini? kertas? Hasil?

https://note.com/mizuiro_shodo/n/n4515759a6ed8

**Comment by vladmandic** [comment]
english only please.

anyhow, this feature request is ok, but it should be done upstream - once its implemented in k-diffusers, sdnext can use it.
sdnext does not have its own fork of k-diffusers nor do i plan to start one, so any direct modifications to k-diffusers code from within sdnext is currently not an option. please create a feature request once this is implemented upstream in k-diffusers.

**Comment by Satriouz** [comment]
Thanks

**Comment by strouder** [comment]
Why is DPM++ 2M Karras missing? @vladmandic

**Comment by vladmandic** [comment]
> Why is DPM++ 2M Karras missing? @vladmandic

its NOT.

DPM++ 2M is a scheduler.
Karras is a variation. One of many. You can choose to run DPM++ 2M in default or karras or several other variations.
Concept that "*** karras" are separate schedulers was wrong from beggining and time to let it go - especially now that we have soo many variations that having a flat list of schedulers would result in 60+ schedulers.
